watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,495 @@
+
+
+def display_model_metrics(m):
+
+    print("---Flow properties in feed, product and disposal---")
+
+    fp = {
+        "Feed": [
+            value(m.fs.feed.properties[0].flow_vol_phase["Liq"]),
+            value(
+                sum(
+                    m.fs.feed.properties[0].conc_mass_phase_comp["Liq", j]
+                    for j in m.fs.properties.ion_set
+                )
+            ),
+        ],
+        "Product": [
+            value(m.fs.prod.properties[0].flow_vol_phase["Liq"]),
+            value(m.fs.prod_salinity),
+        ],
+        "Disposal": [
+            value(m.fs.disp.properties[0].flow_vol_phase["Liq"]),
+            value(m.fs.disp_salinity),
+        ],
+    }
+    fp_table = pd.DataFrame(
+        data=fp,
+        index=["Volume Flow Rate (m3/s)", "Total Ion Mass Concentration (kg/m3)"],
+    )
+    print(fp_table)
+
+    print("---Performance Metrics---")
+
+    pm_table = pd.DataFrame(
+        data=[
+            value(m.fs.recovery_vol_H2O),
+            value(m.fs.mem_area),
+            value(m.fs.EDstack.cell_pair_num),
+            value(m.fs.EDstack.cell_length),
+            value(m.fs.EDstack.voltage_applied[0]),
+            value(m.fs.costing.specific_energy_consumption),
+            value(m.fs.costing.LCOW),
+        ],
+        columns=["value"],
+        index=[
+            "Water recovery by volume",
+            "Total membrane area (aem or cem), m2",
+            "ED cell pair number",
+            "ED cell flow path length",
+            "Operation Voltage, V",
+            "Specific energy consumption, kWh/m3",
+            "Levelized cost of water, $/m3",
+        ],
+    )
+    print(pm_table)
+    print("---Pressure and Temperature point checking---")
+
+    pt_dict = {
+        "Feed": (
+            value(m.fs.feed.outlet.pressure[0]),
+            value(m.fs.feed.outlet.temperature[0]),
+        ),
+        "ED_in_dil": (
+            value(m.fs.EDstack.inlet_diluate.pressure[0]),
+            value(m.fs.EDstack.inlet_diluate.temperature[0]),
+        ),
+        "ED_in_conc": (
+            value(m.fs.EDstack.inlet_concentrate.pressure[0]),
+            value(m.fs.EDstack.inlet_concentrate.temperature[0]),
+        ),
+        "ED_out_dil": (
+            value(m.fs.EDstack.outlet_diluate.pressure[0]),
+            value(m.fs.EDstack.outlet_diluate.temperature[0]),
+        ),
+        "ED_out_conc": (
+            value(m.fs.EDstack.outlet_concentrate.pressure[0]),
+            value(m.fs.EDstack.outlet_concentrate.temperature[0]),
+        ),
+        "Prod": (
+            value(m.fs.prod.inlet.pressure[0]),
+            value(m.fs.prod.inlet.temperature[0]),
+        ),
+        "Disp": (
+            value(m.fs.disp.inlet.pressure[0]),
+            value(m.fs.disp.inlet.temperature[0]),
+        ),
+    }
+    pt_table = pd.DataFrame(data=pt_dict, index=["Pressure (Pa)", "Temperature (K)"])
+    print(pt_table)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
```suggestion
    m = main()
```
So you can access this if running inside a notebook.

Reply:
accepted.